cli/cmd/inject_util.go

```diff
@@ -48,11 +50,13 @@ func transformInput(inputs []io.Reader, errWriter, outWriter io.Writer, rt resou
 }
 
 // processYAML takes an input stream of YAML, outputting injected/uninjected YAML to out.
-func processYAML(in io.Reader, out io.Writer, report io.Writer, rt resourceTransformer) error {
+func processYAML(in io.Reader, out io.Writer, report io.Writer, rt resourceTransformer) []string {
 	reader := yamlDecoder.NewYAMLReader(bufio.NewReaderSize(in, 4096))
 
 	reports := []inject.Report{}
 
+	errorMessages := []string{}
```

Comment:
Given this slice is filled with calls to `err.Error()`, I think it makes more sense to return a slice of `error`.

Reply:
@alpeb That would indeed make the code cleaner! However, we're using the returned slice of string error message [here](https://github.com/linkerd/linkerd2/blob/fe71f52c63786c85a4456de17633fd09f47cf6f2/cli/cmd/inject_util.go#L35) to concatenate the error strings and throw that as an error message.
If we were to return a slice of strings instead, we might need another helper function that first converts the slice of `error`s to slice of `string`s and then use it in `strings.Join` to produce a final error message. WDYT? :)

Reply:
Ok how about giving a try at using wrapped errors?
https://play.golang.org/p/nqNrM8qOvST
You could have `processYAML()` return just `error`, which would really be wrapped errors, which would not need to be joined on the calling side. I *think* this might result in a more go-ish and elegant way!